File: Sort-Algorithms-2.0/Python/selectionSort.py
import os
import random
from time import time

def seleccion(A,n): 
    i = 0
    j = n - 1
    while(i < j): 
        min = A[i] 
        max = A[i] 
        min_i = i 
        max_i = i 
        for k in range(i, j + 1, 1): 
            if (A[k] > max): 
                max = A[k] 
                max_i = k 
            elif (A[k] < min): 
                min = A[k] 
                min_i = k 
        temp = A[i] 
        A[i] = A[min_i] 
        A[min_i] = temp 
        if (A[min_i] == max): 
            temp = A[j] 
            A[j] = A[min_i] 
            A[min_i] = temp 
        else: 
            temp = A[j] 
            A[j] = A[max_i] 
            A[max_i] = temp 
  
        i += 1
        j -= 1

        
def crear_arreglo(A,n):
    for i in range(n):
        A.append(random.randint(1,1000000))



if __name__=='__main__':
    A=[1000000]
    f=open("tiempos_selection_python.txt","w")
    for i in range(100000,1000001,100000):
        crear_arreglo(A,i)
        tiempo_inicial=time()
        seleccion(A,i)
        tiempo_final=time()
        tiempo_ejecucion=tiempo_final-tiempo_inicial
        f.write(str(tiempo_ejecucion))
        print("tiempo de ejecucion: ",tiempo_ejecucion)
    f.close()

File: Sort-Algorithms-2.0/C++/selectionSort.cpp
#include<iostream>
#include<iomanip>
#include<time.h>
#include<stdlib.h>
#include<fstream>

using namespace std;


void crear_arreglo(int *A, int n){
    int c;
    srand(time(NULL));

    for(c=0;c<=n;c++){
        A[c]=rand()%1000000;

    }
}


void mostrar_arreglo(int v[], int n){
    for(int i=0;i<n;i++){
        cout<<v[i]<<setw(3);
    }
}

void selection(int* A, int n)
{
    for (int i = 0, j = n - 1; i < j; i++, j--) {
        int min = A[i], max = A[i];
        int min_i = i, max_i = i;
        for (int k = i; k <= j; k++)  {
            if (A[k] > max) {
                max = A[k];
                max_i = k;
            } else if (A[k] < min) {
                min = A[k];
                min_i = k;
            }
        }

        swap(A[i], A[min_i]);
        if (A[min_i] == max)
            swap(A[j], A[min_i]);
        else
            swap(A[j], A[max_i]);
    }
}

int main()
{
    unsigned t0,t1;
    double time;
    int *v=new int[1000000];
    ofstream archivo;
    archivo.open("tiempos_selection_c2.txt",ios::out);
    for(int i=100000;i<=1000000;i+=100000){
        crear_arreglo(v,i);
        t0=clock();
        selection(v,i);
        t1=clock();
        time=(double(t1-t0)/CLOCKS_PER_SEC);
        cout<<"Tiempo de ejecucion: "<<time<<endl;
        archivo<<time<<endl;
    }

    archivo.close();
    delete[] v;
    return 0;

}
